jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/extensions/compress/CompressExtension.java

```diff
@@ -199,11 +204,14 @@ protected void decompress(ByteAccumulator accumulator, ByteBuffer buf) throws Da
             }
 
             int read;
-            while ((read = inflater.inflate(output)) >= 0)
+            
+            while (true)
```

Comment:
Don't change the while loop on the inflater.
It exists like it did for a reason. (maximum compatibility with all JVMs we run on).
Focus on the ByteAccumulator only in this PR please.

Reply:
@leonchen83 I am not a big fan of this loop change either, now we have 3 nested while loops and a `loop` label. Can this be done in a simpler way?

Reply:
change while loop is necessary. because we change the `ByteBuffer` allocation logic of `ByteAccumulator`. previous loop logic will raise a bug.
I have reduced the loop logic without `loop` label